src/main/java/com/sngular/kloadgen/sampler/schemaregistry/impl/ApicurioSchemaRegistry.java

```diff
@@ -30,147 +26,130 @@
+        schema.setSchema(parserJson.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
+      } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(searchedArtifactType)) {
+        final SchemaParser<ProtobufSchema, Message> parserProtobuf = new ProtobufSchemaParser<>();
+        schema.setSchema(parserProtobuf.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
       } else {
-        SearchedArtifact searchedArtifact = artifacts.get(0);
-        InputStream inputStream = this.schemaRegistryClient.getLatestArtifact(searchedArtifact.getGroupId(), searchedArtifact.getId());
-        String result = IOUtils.toString(inputStream, String.valueOf(StandardCharsets.UTF_8));
-
-        String searchedArtifactType = searchedArtifact.getType().toString();
-        if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(searchedArtifactType)) {
-          SchemaParser parserAvro = new AvroSchemaParser(null);
-          schema.setSchema(parserAvro.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
-        } else if (SchemaTypeEnum.JSON.name().equalsIgnoreCase(searchedArtifactType)) {
-          SchemaParser parserJson = new JsonSchemaParser();
-          schema.setSchema(parserJson.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
-        } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(searchedArtifactType)) {
-          SchemaParser parserProtobuf = new ProtobufSchemaParser();
-          schema.setSchema(parserProtobuf.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
-        } else {
-          throw new KLoadGenException(String.format("Schema type not supported %s", searchedArtifactType));
-        }
-        schema.setType(searchedArtifactType);
-        ParsedSchemaAdapter parsedSchemaAdapter = schema;
-        return new BaseParsedSchema(parsedSchemaAdapter);
+        throw new KLoadGenException(String.format("Schema type not supported %s", searchedArtifactType));
       }
-    } catch (IOException e) {
+      schema.setType(searchedArtifactType);
+      return new BaseParsedSchema<>(schema);
+    } catch (final IOException e) {
       throw new KLoadGenException(e);
     }
   }
 
-  public BaseParsedSchema<ApicurioParsedSchemaMetadata> getSchemaBySubjectAndId(String subjectName, BaseSchemaMetadata<? extends SchemaMetadataAdapter> metadata) {
+  public final BaseParsedSchema<ApicurioParsedSchemaMetadata> getSchemaBySubjectAndId(
+      final String subjectName, final BaseSchemaMetadata<? extends SchemaMetadataAdapter> metadata) {
     final ApicurioParsedSchemaMetadata schema = new ApicurioParsedSchemaMetadata();
 
-    SchemaMetadataAdapter schemaMetadataAdapter = metadata.getSchemaMetadataAdapter();
+    final SchemaMetadataAdapter schemaMetadataAdapter = metadata.getSchemaMetadataAdapter();
     try {
-      InputStream inputStream = this.schemaRegistryClient.getContentByGlobalId(schemaMetadataAdapter.getGlobalId());
```

Comment:
The same as in the previous comment; and maybe some other refactors would be possible.

Reply:
Done!